### ISSUE TITLE
[$250]  iOS - Composer - Suggestion list is partially blocked by keyboard when composer is expanded

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#45825]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.10-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause internal team
**Slack conversation:** 

## Action Performed:
1. Go to staging.new.expensify.com
2. Go to any chat.
3. Paste the following long message

Open any chat.
Insert more text into the input field than can be displayed.
Expand the main text input field.
Verify that the input field can be expanded and shrunk back.
Ensure that the text input field can be scrolled.
Check if the main chat list is scrollable. Open any chat. 😀
Insert more text into the input field than can be displayed.
Expand the main text input field.
Verify that the input field can be expanded and shrunk back.
Ensure that the text input field can be scrolled.
Check if the main chat list is scrollable. Open any chat.
Insert more text into the input field than can be displayed.
Expand the main text input field.
Verify that the input field can be expanded and shrunk back.
Ensure that the text input field can be scrolled.
Check if the main chat list is scrollable.

4. Expand the composer.
5. Tap on the first line.
6. Add @ and followed by user name.
7. Remove @ and name.
8. Repeat Step 6 and 7 a few times.

## Expected Result:
Suggestion list will appear under the cursor when @ is typed on the first line when the composer is expanded.

## Actual Result:
Suggestion list is blocked partially by the keyboard.

## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/9b9f33aa-3d9f-4291-bb91-79930381d412

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~019572cf1b9a5dff60</li>
        <li>Upwork Job ID: 1815808643215834616</li>
        <li>Last Price Increase: 2024-07-23</li>
<li>Automatic offers: </li>
<ul>
<li>abdulrahuman5196 | Reviewer | 103327376</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @abdulrahuman5196</details>

**Comment by m-natarajan** [comment]
@abekkala FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by m-natarajan** [comment]
We think that this bug might be related to  #vip-vsb

**Comment by abdulrahuman5196** [comment]
We don't have any proposal to review

**Comment by kirillzyusko** [comment]
Hi, I am Kiryl from Margelo, an expert agency and I can work on this issue

**Comment by perunt** [comment]
Hey there, I am Taras, also from the Margelo team, an expert agency
@kirillzyusko, I can pick it up as I was the one who originally implemented the suggestion box and full-screen mode, so I have some context here

**Comment by kirillzyusko** [comment]
@perunt yes, sure 🙌

**Comment by perunt** [comment]
I fixed it here previously the case was that cursor coordinates for the first update was a '0' . I fixed it here
https://github.com/Expensify/App/blob/66ebb3e7818ef9c1ed2a768363c512a918bb233a/src/components/AutoCompleteSuggestions/index.tsx#L136

**Comment by abdulrahuman5196** [comment]
@abekkala Can we add the retest label? I am currently having local setup issues with iOS native.

And on a side note, do we need to consider this as a bug on second thoughts, seems the steps to reproduce are too complex for normal user to face this?

**Comment by abekkala** [comment]
VSB on hold - closing
